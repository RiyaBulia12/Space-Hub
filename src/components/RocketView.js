import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import fetchRockets from '../middleware/api';

const RocketView = () => {
  const rocketsObj = useSelector((state) => state.rockets);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchRockets());
  }, [rocketsObj.length, dispatch]);

  return (
    <>
      { rocketsObj.map((item) => (
        <div key={item.id} className="mx-8 lg:mx-36">
          <div className="flex flex-col bg-white border shadow-md my-8 md:flex-row text-neutral-700">
            <img className="object-cover w-full h-72 md:h-auto md:w-64" src={item.flickr_images} alt={item.rocket_name} />
            <div className="flex flex-col p-4 leading-normal">
              <h5 className="mb-2 text-2xl font-bold">{item.rocket_name}</h5>
              <p className="mb-3 font-normal ">{item.description}</p>
              <button
                type="button"
                className="w-36 px-4 py-2.5 bg-blue-600 text-white text-sm
              rounded shadow-md hover:bg-blue-700 hover:shadow-lg transition duration-150 ease-in-out"
              >
                Reserve Rocket
              </button>
            </div>
          </div>
        </div>
      ))}
    </>
  );
};

export default RocketView;
