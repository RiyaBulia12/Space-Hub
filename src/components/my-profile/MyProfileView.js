import React from 'react';
import JoinedMissions from './JoinedMissions/JoinedMissions';

const MyProfileView = () => (
  <div className="mx-[8%] my-10">
    <JoinedMissions />
  </div>
);

export default MyProfileView;
