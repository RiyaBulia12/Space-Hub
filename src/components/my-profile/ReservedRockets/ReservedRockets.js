import React from 'react';
import { useSelector } from 'react-redux';

const ReservedRockets = () => {
  const reservedRockets = useSelector((state) => state.rockets
    .filter((rocket) => rocket.reserved));

  return (
    <div className="flex-1">
      <h2 className="text-3xl">My Rockets</h2>
      <ul className="mt-3 border-2 w-4/12 rounded">
        {reservedRockets.length > 0
          ? reservedRockets.map((rocket) => (
            <li className="text-lg font-medium p-5 border-b-2 last:border-b-0 border-slate-300" key={rocket.id}>
              {rocket.rocket_name}
            </li>
          ))
          : (
            <span className="px-10 text-lg">
              You haven&apos;t reserved any Rockets yet
            </span>
          )}
      </ul>
    </div>
  );
};

export default ReservedRockets;
