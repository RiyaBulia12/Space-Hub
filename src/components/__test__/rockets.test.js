import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';
import RocketView from '../RocketView';
import ReservedRockets from '../my-profile/ReservedRockets/ReservedRockets';
import  fetchRockets from '../../middleware/RocketApi';
import MyProfileView from '../my-profile/MyProfileView';

const MockRockets = () => (
  <Provider store={store}>
    <RocketView />
  </Provider>
);

describe('Rockets View', () => {

  it('renders rockets page correctly', () => {
    const tree = renderer.create(<MockRockets />);
    expect(tree).toMatchSnapshot();
  });

  it('renders list of rockets correctly', async () => {
    await store.dispatch(fetchRockets());
    expect(store.getState().rockets.length).toEqual(4);
  });

});

const MockReservedRockets = () => (
  <Provider store={store}>
    <ReservedRockets />
  </Provider>
);

describe('Reserved Rockets View', () => {

  it('renders ReservedRockets correctly', () => {
    const tree = renderer.create(<MockReservedRockets />);
    expect(tree).toMatchSnapshot();
  });

});
