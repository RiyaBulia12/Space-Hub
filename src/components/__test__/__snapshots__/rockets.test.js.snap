// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reserved Rockets View renders ReservedRockets correctly 1`] = `
<div
  className="flex-1"
>
  <h2
    className="text-3xl"
  >
    My Rockets
  </h2>
  <ul
    className="mt-3 border-2 w-4/12 rounded"
  >
    <span
      className="px-10 text-lg"
    >
      You haven't reserved any Rockets yet
    </span>
  </ul>
</div>
`;

exports[`Rockets View renders rockets page correctly 1`] = `null`;
